### PR TITLE
Added Fathom Analytics support to the documentation

### PR DESCRIPTION
**Pull request by jfsagasti**
- Updated Docusaurus to 2.3.1.
- Added Fathom Analytics plug-in.
- Added several Docusaurus component wrappers to be able to track on-click events.
- Tracked a first set of defined events (more to come in a second stage).

### CODE REVIEW

**Review by adrian-lorenzo — APPROVED** [review]
Awesome :)

Just one comment: You added a `yarn.lock` file, which doesn't seem to have a reference. What's the origin of this?

File: website/src/theme/DocSidebarItem/index.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import DocSidebarItem from '@theme-original/DocSidebarItem';
+
+export default function DocSidebarItemWrapper(props) {
+  const handleClick = (customProps) => {
+    if (customProps && customProps.trackingEvent) {
+      fathom.trackGoal(customProps.trackingEvent, 0);
+      console.log(`DocSidebarItem clicked with trackingEvent: ${customProps.trackingEvent}`);
```
Comment (BorjaZarco) [inline website/src/theme/DocSidebarItem/index.js:8]:
```suggestion
```

File: website/src/theme/NavbarItem/NavbarNavLink.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import NavbarNavLink from '@theme-original/NavbarItem/NavbarNavLink';
+
+export default function NavbarNavLinkWrapper(props) {
+  const handleClick = (customProps) => {
+    console.log('NavbarNavLinkWrapper clicked', customProps);
+    if (customProps && customProps.trackingEvent) {
+      fathom.trackGoal(customProps.trackingEvent, 0);
+      console.log(`Navbar link clicked with trackingEvent: ${customProps.trackingEvent}`);
```
Comment (BorjaZarco) [inline website/src/theme/NavbarItem/NavbarNavLink.js:9]:
```suggestion
```

**Review by BorjaZarco — APPROVED** [review]
LGTM! 👍 

I would remove the `console.log` if they have no meaning than debugging purposes. They may show up on the user (I don't know if docusaurus hides console.logs on prod)

**Pull request merged**